src/utils/handleOptions.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Evaluates the value of a moldObject property.
+ * If the value is a function, it is called with the filtered options object as the its parameter.
+ * Otherwise, the value is returned as is.
+ * @param {object} filteredObject - the current filtered options object
+ * @param {any} value - the value to evaluate
+ * @returns {any} the evaluated value
+ */
+function evaluate(filteredObject, value) {
+  if (typeof value === "function") {
+    return value(filteredObject);
+  } else {
+    return value;
+  }
+}
+
+/**
+ * Checks if a value is within the enumArray.
+ * If isCaseInsensitive is true, the comparison is case insensitive.
+ *
+ * @param {any} value - the value to check
+ * @param {Array} enumArray - the array of enum values
+ * @param {boolean} isCaseInsensitive - whether the comparison is case insensitive
+ * @returns {any | undefined} the enum value if the user value is within the enumArray, otherwise undefined
+ */
+function checkEnum(value, enumArray, isCaseInsensitive) {
+  if (isCaseInsensitive && typeof value === "string") {
+    for (const enumValue of enumArray) {
+      if (enumValue.toLowerCase() === userValue.toLowerCase()) {
+        return enumValue;
+      }
+    }
+  } else {
+    for (const enumValue of enumArray) {
+      if (enumValue === value) {
+        return enumValue;
+      }
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Checks if a value is within the range of min and max.
+ * @param {number} value - the value to check
+ * @param {number} min - the minimum value
+ * @param {number} max - the maximum value
+ * @returns {any | undefined} the value if it is within the range, otherwise undefined
+ */
+function checkNumber(value, min, max) {
+  if (value < min || value > max) {
+    return undefined;
+  } else {
+    return value;
+  }
+}
+
+/**
+ * This function takes a userObject and a moldObject as parameters.
+ * Filters the properties of userObject based on the rules defined in moldObject and returns a filteredObject.
+ * Logs out friendly warnings if the properties in userObject did not fulfill the rules.
+ *
+ * @param {object} userObject - options object provided by the user
+ * @param {object} moldObject - an object that defines how the user options object should be filtered
+ * @returns {object} - filtered options object
+ */
+function handleOptions(userObject, moldObject) {
+  const filteredObject = {};
+
+  for (const key in moldObject) {
+    const userValue = userObject[key];
+    const rules = moldObject[key];
+    const type = evaluate(filteredObject, rules.type);
+    const defaultValue = evaluate(filteredObject, rules.default);
+
+    if (type === "undefined" || type === undefined) {
+      continue;
```

Comment:
Maybe check if the user provided a value and warn them that it is being ignored?